### DISCUSSION TITLE
Enable API for creating repository environment variables

### DISCUSSION THREAD

**Discussion by MartinQuesada** [community/community#47693] [Actions] [answered]
### Select Topic Area

General

### Body

Hello everyone,
I'm creating a repository template, and I need to create some environment secrets and variables for the repository. GitHub API allows me to create the secrets, but it doesn't work for the variables.
It seems that the API doesn't support that feature. Do you have plans to enable it?

I'm running this command
`curl -X PUT -H "Accept: application/vnd.github+json" -H "Authorization: Bearer XXX" -H "X-GitHub-Api-Version: 2022-11-28" https://api.github.com/repositories/1234567890/environments/dev/variables/TEST -d '{"value":"Test"}'`

Thanks in advance

**Comment by tauhid621** [comment, marked as answer]
👋 @MartinQuesada Repository environment variables can be created using API. Here's the [rest API doc](https://docs.github.com/en/rest/actions/variables?apiVersion=2022-11-28#create-an-environment-variable).

For variables, you need to make a `POST` request with `name` and `value` in the body.

`curl -X POST -H "Accept: application/vnd.github+json" -H "Authorization: Bearer XXX" -H "X-GitHub-Api-Version: 2022-11-28" https://api.github.com/repositories/1234567890/environments/dev/variables-d '{"name":"Variable_name", "value":"Test"}'`

Do note that the name of variable is not part of the url in `POST` request.

**Reply by MartinQuesada** [reply]
It worked!
Thank you!
